packages/typespec-vscode/src/openapi3-preview.ts

```diff
@@ -0,0 +1,251 @@
+import { readdir, readFile, rm } from "fs/promises";
+import { basename, dirname, join } from "path";
+import * as semver from "semver";
+import * as vscode from "vscode";
+import logger from "./log/logger.js";
+import { TspLanguageClient } from "./tsp-language-client.js";
+import { createTempDir, ExecOutput } from "./utils.js";
+
+export async function getMainTspFile(): Promise<string | undefined> {
+  const files = await vscode.workspace.findFiles("**/main.tsp").then((uris) => {
+    return uris.filter((uri) => uri.scheme === "file" && !uri.fsPath.includes("node_modules"));
+  });
+
+  switch (files.length) {
+    case 0:
+      const errMsg = "No 'main.tsp' file found in the workspace.";
+      logger.error(errMsg);
+      vscode.window.showErrorMessage(errMsg);
+      return undefined;
+    case 1:
+      return files[0].fsPath;
+    default:
+      return await vscode.window.showQuickPick(
+        files.map((file) => file.fsPath),
+        {
+          title: "Select the 'main.tsp' file",
+        },
+      );
+  }
+}
+
+const openApi3TempFolders = new Map<string, string>();
+const openApi3PreviewPanels = new Map<string, vscode.WebviewPanel>();
+const selectedOpenApi3OutputFiles = new Map<string, string>();
+
+export async function loadOpenApi3PreviewPanel(
+  mainTspFile: string,
+  context: vscode.ExtensionContext,
+  client: TspLanguageClient,
+) {
+  const compilerVersion = client.initializeResult?.serverInfo?.version;
+  if (!compilerVersion || semver.lt(compilerVersion, "0.65.0")) {
```

Comment:
0.65.0? i think the capacity is added at 0.64.0. Please help to double check.

Reply:
After discussion, we thought it's OK to ask for the latest version, e.g. 0.65.0